titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/sanitizer/JobAssertions.java

```diff
@@ -56,6 +63,20 @@ public boolean isValidIamRole(String iamRole) {
         return !StringExt.safeTrim(iamRole).isEmpty();
     }
 
+    public boolean isEnvironmentNotTooLarge(Map<String, String> environment) {
```

Comment:
Really, we should make a TODO to convert the environment variables to byte arrays, and introduce the following two requirements to them:

- No `.` in the key
- No NULL `\0` in the key nor value

Reply:
We could verify that all characters map correctly to ASCII charset. On top of this, we could also check for `.` and `\0`.

Reply:
right now, people use ".", which is bad, so we'll have to roam and ride.